src/lib/util/sbuff.h

```diff
@@ -113,6 +113,7 @@ typedef struct {
 	FILE			*file;			//!< FILE * we're reading from.
 	char			*buff_end;		//!< The true end of the buffer.
 	size_t			max;			//!< Maximum number of bytes to read.
+	size_t			read;			//!< Number of bytes read.
```

Comment:
Isn't this equivalent to `available + shifted`?

Reply:
I do believe you're correct. I'll make the change, test, and push. Thanks.

Reply:
Turns out that the equivalent is shifted + (end - buff); will change, retest, and push.

Reply:
maybe something like `shifted + fr_sbuff_used(self)` ?

Reply:
Nah @jejones3141 is correct if we're trying to limit the total amount read from a file, and not limit the total amount we've read from the buffer.